File: Server/src/main/java/org/f0w/fproject/server/Application.java
package org.f0w.fproject.server;

import org.elasticsearch.client.Client;
import org.f0w.fproject.server.utils.IOUtils;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.scheduling.annotation.EnableScheduling;

import javax.annotation.PostConstruct;

@SpringBootApplication
@EnableScheduling
public class Application {
    private static final Logger LOG = LoggerFactory.getLogger(Application.class);

    @Autowired
    private Client elastic;

    @PostConstruct
    public void init() {
        if (!elastic.admin().indices().prepareExists(Constants.ELASTIC_FOOD_INDEX).get().isExists()) {
            LOG.info("[Elastic] Создание индекса {}", Constants.ELASTIC_FOOD_INDEX);
            String settings = IOUtils.readFileFromResources("elastic/food.index.json");
            elastic.admin()
                .indices()
                .prepareCreate(Constants.ELASTIC_FOOD_INDEX)
                .addMapping(Constants.FOOD, settings)
                .get();
        }
    }

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }
}

File: Server/src/main/java/org/f0w/fproject/server/Application.kt
package org.f0w.fproject.server

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.dataformat.csv.CsvMapper
import com.squareup.okhttp.OkHttpClient
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Primary
import org.springframework.scheduling.annotation.EnableScheduling
import org.yaml.snakeyaml.Yaml

@SpringBootApplication
@EnableScheduling
open class Application {
    @Bean
    open fun yaml() = Yaml()

    @Bean
    open fun httpClient() = OkHttpClient()

    @Bean
    @Primary
    open fun jsonMapper(): ObjectMapper {
        val jsonMapper = ObjectMapper()
        jsonMapper.findAndRegisterModules()

        return jsonMapper
    }

    @Bean
    open fun csvMapper(): CsvMapper {
        val csvMapper = CsvMapper()
        csvMapper.findAndRegisterModules()

        return csvMapper
    }
}

fun main(args: Array<String>) {
    SpringApplication.run(Application::class.java, *args)
}